app/src/RelayEnvironment.ts

```diff
@@ -39,7 +40,20 @@ function fetchJsonObservable<T>(
     const controller = new AbortController();
 
     graphQLFetch(input, { ...init, signal: controller.signal })
-      .then((response) => response.json())
+      .then((response) => {
+        // handle redirects from the server if any
+        if (response.status === 307) {
+          // If the server returns a 307, we should redirect to the login page
+          window.location.href = response.headers.get("Location") ?? "";
```

Comment:
Slightly feels like this belongs more in authFetch but I get it.

Fallback to "" seems sorta non ideal. Maybe use invariant here too?